apps/coordinator/src/components/Wallet/TransactionsTab/FeeBumping/hooks/useFeeBumping.ts

```diff
@@ -0,0 +1,265 @@
+   *
+   * @param tx - The transaction object to analyze
+   * @param rawTxHex - The raw transaction hex string
+   */
+  const analyzeTx = useCallback(
+    async (
+      tx: any,
+      initialTxHex: string = "",
+      priority: FeePriority = FeePriority.MEDIUM,
+    ) => {
+      if (!tx) return;
+      try {
+        feeBumpDispatch(setFeeBumpStatus(FeeBumpStatus.ANALYZING));
+        feeBumpDispatch(setFeeBumpError(null));
+
+        // Get blockchain client
+        const blockchainClient = dispatch(
+          updateBlockchainClient(),
+        ) as unknown as BlockchainClient;
+        if (!blockchainClient) {
+          throw new Error("Blockchain client not available");
+        }
+
+        // Extract UTXOs for fee bumping
+        const availableUtxos = await extractUtxosForFeeBumping(
+          tx,
+          depositNodes,
+          changeNodes,
+          blockchainClient,
+        );
+
+        if (!availableUtxos.length) {
+          throw new Error("No UTXOs available for fee bumping");
+        }
+        const walletConfig = {
+          requiredSigners,
+          totalSigners,
+          addressType,
+        };
+
+        // Analyze transaction for fee bumping options
+        const analysis = await analyzeTransaction(
+          initialTxHex,
+          tx.fee, // Fee in satoshis
+          network,
+          availableUtxos,
+          blockchainClient,
+          walletConfig,
+          priority,
+        );
+        // We add some additional UI-friendly details which we'll use to show the user
+        const feeBumpRecommendation: FeeBumpRecommendation = {
+          ...analysis,
+          // Add more user-friendly fields
+          currentFeeRate: analysis.feeRate,
+          canRBF: analysis.canRBF,
+          canCPFP: analysis.canCPFP,
+          suggestedRBFFeeRate: Math.max(
+            analysis.userSelectedFeeRate,
+            Number(analysis.estimatedRBFFee) / analysis.vsize, // Is this correct ??
+            // Noting that the new tx would have a size greater than original so at worst it gives a higher feeRate which is good
+          ),
+          suggestedCPFPFeeRate: Math.max(
+            analysis.userSelectedFeeRate,
+            Number(analysis.estimatedCPFPFee) / analysis.vsize, //  Is this correct ??
+          ),
+        };
+
+        feeBumpDispatch(setFeeBumpRecommendation(feeBumpRecommendation));
+        feeBumpDispatch(setFeeBumpStatus(FeeBumpStatus.READY));
+      } catch (error) {
+        console.error("Error analyzing transaction:", error);
+        feeBumpDispatch(
+          setFeeBumpError(
+            error instanceof Error
+              ? error.message
+              : "Unknown error analyzing transaction",
+          ),
+        );
+      }
+    },
+    [
+      dispatch,
+      feeBumpDispatch,
+      network,
+      requiredSigners,
+      totalSigners,
+      depositNodes,
+      changeNodes,
+      addressType,
```

Comment:
these could be compressed down to just `walletConfig` if you have the single selector. 
```suggestion
      depositNodes,
      changeNodes,
      walletConfig,
```